includes/class-wsuwp-vals-custom-roles.php

```diff
@@ -26,5 +47,588 @@ public static function get_instance() {
 	 *
 	 * @since 0.0.1
 	 */
-	public function setup_hooks() {}
+	public function setup_hooks() {
+		add_action( 'init', array( $this, 'register_taxonomy' ), 12 );
+		add_action( 'personal_options', array( $this, 'extend_user_profile' ) );
+		add_action( 'personal_options_update', array( $this, 'save_user_center_data' ) );
+		add_action( 'edit_user_profile_update', array( $this, 'save_user_center_data' ) );
+		add_action( 'admin_menu', array( $this, 'center_admin_page' ) );
+		add_filter( 'parent_file', array( $this, 'user_center_page' ) );
+		add_filter( 'manage_edit-center_columns', array( $this, 'center_user_column' ) );
+		add_action( 'manage_center_custom_column', array( $this, 'manage_center_column' ), 10, 3 );
+		add_filter( 'manage_users_columns', array( $this, 'users_custom_columns' ) );
+		add_action( 'manage_users_custom_column', array( $this, 'manage_users_vals_columns' ), 10, 3 );
+		add_filter( 'manage_users_sortable_columns', array( $this, 'manage_users_vals_sortable_columns' ) );
+		add_filter( 'login_redirect', array( $this, 'vals_roles_login_redirect' ), 10, 3 );
+		add_action( 'current_screen', array( $this, 'vals_roles_redirect' ) );
+		add_action( 'admin_init', array( $this, 'vals_roles_menu_pages' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'vals_roles_enqueue_scripts' ) );
+		add_action( 'pre_get_users', array( $this, 'vals_pre_user_query' ) );
+		add_filter( 'views_users', array( $this, 'vals_center_admin_views_users' ) );
+		add_filter( 'editable_roles', array( $this, 'vals_center_admin_editable_roles' ) );
+	}
+
+	/**
+	 * Add 'VALS Registered Trainee' and 'VALS Center Admin' custom roles.
+	 *
+	 * @since 0.0.1
+	 */
+	static function add_roles() {
+		add_role(
+			WSUWP_VALS_Custom_Roles()->roles['trainee'],
+			'VALS Registered Trainee',
+			array(
+				'read' => true,
+			)
+		);
+
+		add_role(
+			WSUWP_VALS_Custom_Roles()->roles['certified'],
+			'VALS Certified',
+			array(
+				'read' => true,
+			)
+		);
+
+		add_role(
+			WSUWP_VALS_Custom_Roles()->roles['admin'],
+			'VALS Center Admin',
+			array(
+				'read' => true,
+				'list_users' => true,
+				'promote_users' => true,
+			)
```

Comment:
Ahh, that's a good point. If someone else (site admin) is adding the users initially as VALS types, then we shouldn't have to worry about moving from built in WP roles -> VALS roles.